src/hotspot/share/opto/predicates.hpp

```diff
@@ -361,26 +361,19 @@ class RuntimePredicate : public Predicate {
   static bool is_predicate(const Node* node, Deoptimization::DeoptReason deopt_reason);
 };
 
-// Class to represent a Template Assertion Predicate.
-class TemplateAssertionPredicate : public Predicate {
+class CommonAssertionPredicate : public Predicate {
   IfTrueNode* const _success_proj;
   IfNode* const _if_node;
 
- public:
-  explicit TemplateAssertionPredicate(IfTrueNode* success_proj)
-      : _success_proj(success_proj),
-        _if_node(success_proj->in(0)->as_If()) {
-    assert(is_predicate(success_proj), "must be valid");
-  }
+public:
+  explicit CommonAssertionPredicate(IfTrueNode* success_proj)
```

Comment:
@chhagedorn might have some reservations about this, but I'll let him comment on his own.

Reply:
The Assertion Predicate code is changing at the moment. I've been waiting with some patches until the fork. There is more coming now. How urgent is the work of this patch? Otherwise, it might be easier to wait for the changes within the next weeks.